src/medCore/gui/toolboxes/medToolBox.cpp

```diff
@@ -308,3 +308,30 @@ void medToolBox::toXMLNode(QDomDocument* doc, QDomElement* currentNode)
 	currentNode->appendChild(elmt);
 }
 
+void medToolBox::handleDisplayError(int error)
+{
+    switch (error)
+    {
+    case medAbstractProcess::PIXEL_TYPE:   //! Handle volume errors: pixel type
+        displayMessageError("Pixel type not yet implemented");
+        break;
+    case medAbstractProcess::DIMENSION_3D: //! Handle volume errors: dimension
+        displayMessageError("This toolbox is designed to be used with 3D volumes");
+        break;
+    case medAbstractProcess::DIMENSION_4D: //! Handle volume errors: dimension
+        displayMessageError("This toolbox is designed to be used with 4D volumes");
+        break;
+    case medAbstractProcess::MESH_TYPE:    //! Handle mesh errors: data type
+        displayMessageError("This toolbox is designed to be used with meshes");
```

Comment:
This case is actually quite restrictive, we cannot use it very often.
Maybe, we could change it to "This toolbox requires (at least) one mesh" or something like that.